frontend/middleware/admin-only.ts

```diff
@@ -0,0 +1,10 @@
+interface AuthRedirectParams {
+    $auth: any
+    redirect: (path: string) => void
+}
+export default function ({ $auth, redirect }: AuthRedirectParams) {
+    // If the user is not an admin redirect to the home page
+    if (!$auth.user.admin) {
```

Comment:
I feel like this shouldn't work for public users (i.e. `$auth.user` is null) but it does. Maybe because it evaluated the `auth` middleware first?

Reply:
yeah, tested that too. 
Im having a thing in the back of my mind that because the auth middleware is defined in the nuxt config it will be evaluated first. 
I'll recheck that tomorrow.

Reply:
Seems to work fine, I just found it weird that it worked

Reply:
by default they will be executed in [alphabetic order](https://nuxt.com/docs/guide/directory-structure/middleware#ordering-global-middleware)